src/command_executor.cc

```diff
@@ -926,6 +927,31 @@ sql_callback(exec_context& ec, sqlite3_stmt* stmt)
             dls.push_header(colname, type, graphable);
             if (graphable) {
                 auto attrs = vc.attrs_for_ident(colname);
+                dls.dls_headers[lpc].hm_colorstr = colname;
+                for (
+                    std::string suf = " ";
+                    std::any_of(
+                        dls.fg_colors_vec.cend() - ( std::min((int)dls.fg_colors_vec.size(), (int) (view_colors::HI_COLOR_COUNT - 1))),
+                        dls.fg_colors_vec.cend(),
+                        [attrs](auto el) {return el == attrs.ta_fg_color;}
+                        ) ||
+                    std::any_of(
+                        dls.bg_colors_vec.cend() - ( std::min((int)dls.bg_colors_vec.size(), (int) (view_colors::HI_COLOR_COUNT - 1))),
+                        dls.bg_colors_vec.cend(),
+                        [attrs](auto el) {return el == attrs.ta_bg_color;}
+                    );
+                    suf += " "
+                ) {
+                    attrs = vc.attrs_for_ident(colname + suf);
+                    dls.dls_headers[lpc].hm_colorstr = colname + suf;
+                }
+
+                if (attrs.ta_fg_color) {
+                    dls.fg_colors_vec.emplace_back(attrs.ta_fg_color);
+                }
+                if (attrs.ta_bg_color) {
+                    dls.bg_colors_vec.emplace_back(attrs.ta_bg_color);
+                }
```

Comment:
Is there a reason to keep these as separate vectors?

Reply:
Mmm my lack of ease with C++ I'd say XD.
You have to consider that I had not touched or read through C++ code in a serious manner for 20 years or so until one or two weeks ago when I started looking through the code of `lnav`, so I am mostly parroting parts of code I see in the codebase or using my knowledge of other languages to look for corresponding structures/syntax to use. It's an interesting journey but sometimes I don't make the best or the most idiomatic choices...

Reply:
I'm not expecting you to know C++ seriously.  I'm trying to understand if there is a semantic reason you broke out the foreground and background color values from the `text_attrs` struct.